src/main/java/bern/functionalities/Parser.java

```diff
@@ -0,0 +1,165 @@
+package bern.functionalities;
+
+import bern.model.Deadline;
+import bern.model.Event;
+import bern.model.Task;
+import bern.model.ToDo;
+
+/**
+ * A class to contain all methods parsing related.
+ */
+public class Parser {
+    /**
+     * A method to create a ToDo instance from a string command.
+     *
+     * @param s The command from which a ToDo instance will be made.
+     * @return The ToDo instance.
+     */
+    public ToDo toDoFromString(String s) {
+        boolean isDone = s.substring(4, 5).equals("X");
+        String desc = s.substring(7);
+        ToDo ans = new ToDo(desc);
+        if (isDone) {
+            ans.markAsDone();
+        }
+        return ans;
+    }
+
+    /**
+     * A method to create a Deadline instance from a string command.
+     *
+     * @param s The command from which a Deadline instance will be made.
+     * @return The Deadline instance.
+     */
+    public Deadline deadlineFromString(String s) {
+        boolean isDone = s.substring(4, 5).equals("X");
+        String desc = s.substring(7, s.indexOf("(by: ") - 1);
+        String by = s.substring(s.indexOf("(by: ") + 5, s.length() - 1);
+        Deadline ans = new Deadline(desc, by);
+        if (isDone) {
+            ans.markAsDone();
+        }
+        return ans;
+    }
```

Comment:
It seems like`deadlineFromString()` is not instance specific. Perhaps tagging it as static makes more sense? Not really a coding style suggestion but thought it might be useful. The same could be applied for the other factory methods as well :)